documentation/design-docs/dotnet-monitor.yaml

```diff
@@ -0,0 +1,472 @@
+  version: "1.0.0"
+  title: "dotnet-monitor"
+host: "localhost:52323"
+basePath: "/diag/v1/"
+schemes:
+- "http"
+paths:
+  /processes:
+    get:
+      tags:
+      - "processes"
+      operationId: "getprocesses"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            type: array
+            items: 
+              $ref: "#/definitions/Process"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /dump:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getdefaultprocessdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "type"
+        in: query
+        description: "Specifies the dump type, which determines the kinds of information that are collected from the process."
+        type: string
+        enum:
+        - "Normal"
+        - "WithHeap"
+        - "Triage"
+        - "Full"
+        required: false
+        default: "Normal"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /dump/{uid}:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getprocessdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      - name: "type"
+        in: query
+        description: "Specifies the dump type, which determines the kinds of information that are collected from the process."
+        type: string
+        enum:
+        - "Normal"
+        - "WithHeap"
+        - "Triage"
+        - "Full"
+        required: false
+        default: "Normal"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /gcdump:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getdefaultgcdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "timeout"
+        in: "query"
+        description: "Maximum duration of the GC Dump"
+        required: false
+        type: "integer"
+        default: "30"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /gcdump/{uid}:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getgcdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "timeout"
+        in: "query"
+        description: "Maximum duration of the GC Dump"
```

Comment:
I would add the unit of time to the description, for example, "Maximum duration of the GC dump in seconds."